src/panels/config/automation/trigger/types/ha-automation-trigger-time.ts

```diff
@@ -82,15 +97,26 @@ export class HaTimeTrigger extends LitElement implements TriggerElement {
       return nothing;
     }
 
-    const inputMode =
-      this._inputMode ??
-      (at?.startsWith("input_datetime.") || at?.startsWith("sensor."));
+    const entity =
+      typeof at === "object"
+        ? at.entity_id
+        : at?.startsWith("input_datetime.") || at?.startsWith("sensor.")
+          ? at
+          : undefined;
+    const time = entity ? undefined : at;
+    const offset = typeof at === "object" ? at.offset : undefined;
+
+    const mode = this._inputMode ?? (entity ? MODE_ENTITY : MODE_TIME);
 
-    const schema = this._schema(this.hass.localize, inputMode);
+    const showOffset = mode === MODE_ENTITY && entity?.startsWith("sensor.");
+
+    const schema = this._schema(this.hass.localize, mode, !!showOffset);
 
     const data = {
-      mode: inputMode ? "input" : "value",
-      ...this.trigger,
+      mode,
+      entity,
+      time,
+      offset,
     };
```

Comment:
We should make data a memoized function so it wont create a new object on every render

Reply:
done